src/python/grpcio_tests/tests/observability/_open_telemetry_observability_test.py

```diff
@@ -0,0 +1,232 @@
+
+        self._validate_metrics_exist(self.all_metrics)
+
+    def testRecordStreamStream(self):
+        otel_plugin = BaseTestOpenTelemetryPlugin(self._provider)
+
+        with grpc_observability.OpenTelemetryObservability(
+            plugins=[otel_plugin]
+        ):
+            server, port = _test_server.start_server()
+            self._server = server
+            _test_server.stream_stream_call(port=port)
+
+        self._validate_metrics_exist(self.all_metrics)
+
+    def testTargetAttributeFilter(self):
+        # If target label contains 'localhost', is should be replaced with 'other'.
+        def target_filter(target: str) -> bool:
+            if "localhost" in target:
+                return False
+            return True
+
+        otel_plugin = BaseTestOpenTelemetryPlugin(self._provider)
+        otel_plugin.target_attribute_filter = target_filter
+
+        with grpc_observability.OpenTelemetryObservability(
+            plugins=[otel_plugin]
+        ):
+            server, port = _test_server.start_server()
+            self._server = server
+            _test_server.unary_unary_call(port=port)
+
+        self._validate_metrics_exist(self.all_metrics)
+        target_values = self.all_metrics["label"].get(GRPC_TARGET_LABEL)
+        self.assertTrue(GRPC_OTHER_LABEL_VALUE in target_values)
+
+    def testMethodAttributeFilter(self):
+        # If method label contains 'UnaryUnaryFiltered', is should be replaced with 'other'.
+        FILTERED_METHOD_NAME = "UnaryUnaryFiltered"
+
+        def method_filter(method: str) -> bool:
+            if FILTERED_METHOD_NAME in method:
+                return False
+            return True
+
+        otel_plugin = BaseTestOpenTelemetryPlugin(self._provider)
+        otel_plugin.generic_method_attribute_filter = method_filter
+
+        with grpc_observability.OpenTelemetryObservability(
+            plugins=[otel_plugin]
+        ):
+            server, port = _test_server.start_server()
+            self._server = server
+            _test_server.unary_unary_call(port=port)
+            _test_server.unary_unary_filtered_call(port=port)
+
+        self._validate_metrics_exist(self.all_metrics)
+        method_values = self.all_metrics["label"].get(GRPC_METHOD_LABEL)
+        self.assertTrue(GRPC_OTHER_LABEL_VALUE in method_values)
+        self.assertTrue(FILTERED_METHOD_NAME not in method_values)
+
+    def _validate_metrics_exist(self, all_metrics: Dict[str, Any]) -> None:
+        # Sleep here to make sure we have at least one export from OTel MetricExporter.
+        time.sleep(OTEL_EXPORT_INTERVAL_S)
+        self.assertGreater(len(all_metrics["name"]), 0)
+        self._validate_metrics_names(all_metrics["name"])
+
+    def _validate_metrics_names(self, metric_names: Set[str]) -> None:
+        for base_metric in _open_telemetry_measures.base_metrics():
+            if base_metric.name not in metric_names:
+                logger.error(
```

Comment:
[`assertTrue` takes a second argument](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertTrue) for a message to print to `stderr`